src/tests/unit/tests/DetailsView/components/save-assessment-button.test.tsx

```diff
@@ -1,47 +1,104 @@
 import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
+import { UserConfigMessageCreator } from 'common/message-creators/user-config-message-creator';
+import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
 import {
     SaveAssessmentButton,
     SaveAssessmentButtonProps,
 } from 'DetailsView/components/save-assessment-button';
-import { shallow } from 'enzyme';
+import { shallow, ShallowWrapper } from 'enzyme';
 import * as React from 'react';
 import { EventStubFactory } from 'tests/unit/common/event-stub-factory';
-import { IMock, It, Mock, MockBehavior } from 'typemoq';
+import { IMock, Mock, Times } from 'typemoq';
 
 describe('SaveAssessmentButton', () => {
     let propsStub: SaveAssessmentButtonProps;
     let detailsViewActionMessageCreatorMock: IMock<DetailsViewActionMessageCreator>;
+    let eventStub: any;
+    let userConfigMessageCreatorMock: IMock<UserConfigMessageCreator>;
+    let userConfigurationStoreData: UserConfigurationStoreData;
 
     beforeEach(() => {
-        detailsViewActionMessageCreatorMock = Mock.ofType(
-            DetailsViewActionMessageCreator,
-            MockBehavior.Strict,
-        );
+        detailsViewActionMessageCreatorMock = Mock.ofType<DetailsViewActionMessageCreator>();
+        userConfigurationStoreData = {
+            showSaveAssessmentDialog: true,
+        } as UserConfigurationStoreData;
+        userConfigMessageCreatorMock = Mock.ofType(UserConfigMessageCreator);
         propsStub = {
-            deps: { detailsViewActionMessageCreator: detailsViewActionMessageCreatorMock.object },
+            deps: {
+                detailsViewActionMessageCreator: detailsViewActionMessageCreatorMock.object,
+                userConfigMessageCreator: userConfigMessageCreatorMock.object,
+            },
             download: 'download',
             href: 'url',
+            userConfigurationStoreData,
         };
+        eventStub = new EventStubFactory().createMouseClickEvent();
     });
 
-    it('should render per the snapshot', () => {
-        const rendered = shallow(<SaveAssessmentButton {...propsStub} />);
+    describe('on dialog enabled', () => {
+        let wrapper: ShallowWrapper;
 
-        expect(rendered.getElement()).toMatchSnapshot();
+        beforeEach(() => {
+            wrapper = shallow(<SaveAssessmentButton {...propsStub} />);
+            wrapper.find(InsightsCommandButton).simulate('click', eventStub);
+        });
+
+        it('renders when dialog is enabled', () => {
+            expect(wrapper.getElement()).toMatchSnapshot();
+        });
+
+        it('is dismissed when "got it" button is clicked', () => {
+            wrapper.find(PrimaryButton).simulate('click');
+
+            expect(wrapper.getElement()).toMatchSnapshot();
```

Comment:
I think it would be preferable to separate the behavioral tests to avoid also testing rendering behavior (ie, have this check for the presence of a `Dialog` child rather than matching a snapshot). If this ever broke in the future, it would be really easy for a reviewer to skim past an important snapshot change.